keras_cv/models/resnet_v2.py

```diff
@@ -99,210 +101,312 @@
+        self.bn1 = layers.BatchNormalization(
+            axis=BN_AXIS, epsilon=BN_EPSILON, name=self.name + "_1_bn"
+        )
+        self.activation1 = layers.Activation("relu", name=self.name + "_1_relu")
+        self.conv2 = layers.Conv2D(
+            self.filters,
+            self.kernel_size,
+            strides=self.stride2,
             padding="same",
-            dilation_rate=dilation,
-            name=name + "_2_conv",
-        )(x)
-        x = layers.BatchNormalization(
-            axis=BN_AXIS, epsilon=1.001e-5, name=name + "_2_bn"
-        )(x)
-        x = layers.Activation("relu", name=name + "_2_relu")(x)
-
-        x = layers.Conv2D(4 * filters, 1, name=name + "_3_conv")(x)
-        x = layers.Add(name=name + "_out")([shortcut, x])
-        return x
-
-    return apply
+            dilation_rate=self.dilation,
+            use_bias=False,
+            name=self.name + "_2_conv",
+        )
+        self.bn2 = layers.BatchNormalization(
+            axis=BN_AXIS, epsilon=BN_EPSILON, name=self.name + "_2_bn"
+        )
+        self.activation2 = layers.Activation("relu", name=self.name + "_2_relu")
+        self.conv3 = layers.Conv2D(
+            4 * filters, kernel_size=1, name=self.name + "_3_conv"
+        )
+        self.add = layers.Add(name=self.name + "_out")
+
+    def call(self, inputs):
+        x = self.bn_preactivation(inputs)
+        x = self.relu_preactivation(x)
+        shortcut = self.shortcut(x) if self.shortcut else x
+        x = self.conv1(x)
+        x = self.bn1(x)
+        x = self.activation1(x)
+        x = self.conv2(x)
+        x = self.bn2(x)
+        x = self.activation2(x)
+        x = self.conv3(x)
+        return self.add([x, shortcut])
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "filters": self.filters,
+                "kernel_size": self.kernel_size,
+                "stride": self.stride,
+                "dilation": self.dilation,
+                "conv_shortcut": self.conv_shortcut,
+            }
+        )
+        return config
 
 
-def Stack(
-    filters,
-    blocks,
-    stride=2,
-    dilations=1,
-    name=None,
-    block_fn=Block,
-    first_shortcut=True,
-    stack_index=1,
-):
+@keras.utils.register_keras_serializable(package="keras_cv.models.resnet_v2")
+class Stack(keras.layers.Layer):
     """A set of stacked blocks.
+
     Args:
         filters: integer, filters of the layer in a block.
         blocks: integer, blocks in the stacked blocks.
-        stride: default 2, stride of the first layer in the first block.
-        name: string, stack label.
+        stride: integer, stride of the first layer in the first block. Defaults to 2.
+        dilation: integer, the dilation rate to use for dilated convolution. Defaults to 1.
         block_fn: callable, `Block` or `BasicBlock`, the block function to stack.
         first_shortcut: default True, use convolution shortcut if True,
```

Comment:
boolean, whether to use a convolutional shortcut in the first block. Defaults to True.